### PR TITLE
[runtime] Get rid of the GLIB_LIBS autoconf define, it prevents automake from detecting the dependency on libeglib.la.

### LINKED ISSUE #8175: Changing eglib doesn't rebuild mono
## Steps to Reproduce

1. `CC='ccache cc' CXX='ccache c++' ./autogen.sh --prefix=${PREFIX} --disable-nls --enable-native-types --enable-debug --enable-checked-build=all --disable-btls --disable-btls-lib`
2. `make -j`
3. `touch mono/eglib/gmarkup.c`
4. `make`

Note that only `eglib` is rebuilt, but the mono binaries and libraries in `mono/mini/` are not.

### On which platforms did you notice this

[x] macOS
[ ] Linux
[ ] Windows

**Version Used**: mono master@ 920267df91abcd7b79a45e81b634d7c107986b09

### PR DESCRIPTION
**Pull request by vargaz**
Fixes https://github.com/mono/mono/issues/8175

### CODE REVIEW

**Comment by luhenry** [comment]
@vargaz could you please use the full URL to the Issues/Bugzilla you are fixing, it makes it easier to figure out if it's a PR, an Issue or a bug when writing release notes. Thank you.

**Comment by lambdageek** [comment]
There are a couple of places in `mono/metadata/Makefile.am` that hardcode `../eglib/libeglib.la` - should probably use a `$(glib_libs)` variable, too.

**Comment by vargaz** [comment]
Will try that.

File: mono/tests/Makefile.am [hunk 1]
```diff
@@ -2417,7 +2417,7 @@ else
 libtest_la_LDFLAGS = -rpath `pwd`
 endif
 libtest_la_SOURCES = libtest.c
-libtest_la_LIBADD = $(GLIB_LIBS)
+libtest_la_LIBADD = $(top_builddir)/mono/eglib/libeglib.la -lm
```
Comment (luhenry) [inline mono/tests/Makefile.am]: leftover `-lm`?

File: mono/unit-tests/Makefile.am [hunk 2]
```diff
@@ -19,10 +19,11 @@ else
 LLVMMONOF=$(LLVM_LIBS) $(LLVM_LDFLAGS)
 endif
 
+glib_libs = $(top_builddir)/mono/eglib/libeglib.la -lm
```
Comment (luhenry) [inline mono/unit-tests/Makefile.am]: leftover `-lm`?

**Comment by luhenry** [comment]
@monojenkins build failed

File: configure.ac [hunk 3]
```diff
@@ -834,7 +834,6 @@ AC_SUBST([WERROR_CFLAGS])
 AC_SUBST([SHARED_CFLAGS])
 
 GLIB_CFLAGS='-I$(top_srcdir)/mono/eglib -I$(top_builddir)/mono/eglib'
-GLIB_LIBS='$(top_builddir)/mono/eglib/libeglib.la -lm'
   
 AC_SUBST(GLIB_CFLAGS)
 AC_SUBST(GLIB_LIBS)
```
Comment (luhenry) [inline configure.ac]: This has to be removed

File: tools/monograph/Makefile.am [hunk 4]
```diff
@@ -34,8 +34,7 @@ AM_CPPFLAGS = 				\
 monograph_LDADD = \
 	$(runtime_lib)			\
 	$(GLIB_LIBS)			\
```
Comment (luhenry) [inline tools/monograph/Makefile.am]: This needs to be modified as well.

**Comment by luhenry** [comment]
@monojenkins build Linux x64

**Pull request merged**